sky/clouds/service_catalog/data_fetchers/fetch_aws.py

```diff
@@ -133,10 +134,12 @@ def _get_availability_zones(region: str) -> Optional[pd.DataFrame]:
         elif e.response['Error']['Code'] == 'UnauthorizedOperation':
             with ux_utils.print_exception_no_traceback():
                 raise RuntimeError(
-                    'Failed to retrieve availability zone. '
+                    'Failed to retrieve availability zones. '
                     'Please ensure that the `ec2:DescribeAvailabilityZones` '
                     'action is enabled for your AWS account in IAM. '
-                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html'  # pylint: disable=line-too-long
+                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html.\n'  # pylint: disable=line-too-long
```

Comment:
Some more detail on this raise would be good - currently just recieve the error: "Details: [botocore.exceptions.ClientError] An error occurred (UnauthorizedOperation) when calling the DescribeAvailabilityZones operation: You are not authorized to perform this operation."

despite: `aws ec2 describe-availability-zones` working fine.

Can you log the account at this point also?

Reply:
Thanks for the comment @TRT-BradleyB! The upper-level caller in the `aws.py` should log the identity information above. Do you see that message when running `sky check`?

Reply:
Hmm, this is weird. I did see the following messages when running `sky check`:
```console
$ sky check                                                     
Checking credentials to enable clouds for SkyPilot.
  Checking AWS...I 08-24 10:09:34 aws_catalog.py:79] Fetching availability zones mapping for AWS...
  AWS: disabled          
    Reason: Failed to fetch the availability zones for the account ['my-account', 'my-account']. It is likely due to permission issues, please check the minimal permission required for AWS: https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/aws.html
    Details: [builtins.RuntimeError] Failed to retrieve availability zones. Please ensure that the `ec2:DescribeAvailabilityZones` action is enabled for your AWS account in IAM. Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html. Details: [botocore.exceptions.ClientError] ...
```

Just to confirm, is the error you shown from `sky check`?

Reply:
Thanks for confirming! Merging this for now, and we can continue the discussion in #2451 for the availability zones permission issue.